python/ambassador/envoy/v3/v3listener.py

```diff
@@ -974,13 +1025,22 @@ def finalize_http(self) -> None:
             self._filter_chains.append(filter_chain)
 
     def as_dict(self) -> dict:
-        listener = {
+        listener: dict = {
             "name": self.name,
             "address": self.address,
+            "udp_listener_config": {},
             "filter_chains": self._filter_chains,
             "traffic_direction": self.traffic_direction
         }
 
+        if self.isProtocolUDP():
+            listener['udp_listener_config'] = {
+                'quic_options': {},
+                'downstream_socket_config': { 'prefer_gro': True }
+            }
+        else:
+            del(listener['udp_listener_config'])
+
```

Comment:
I'm not sure what's to like with this pattern of "initialize with a bogus value, then either overwrite with a valid value or remove it".  Why not drop the initial value and drop the `else` case?

Reply:
agree that is cleaner